server/jobs/moss.py

```diff
@@ -0,0 +1,123 @@
+import os
+import subprocess
+import shlex
+import glob
+import tempfile
+import ntpath
+import re
+
+from server.models import Assignment, Backup, db
+from server.utils import encode_id
+from server import jobs
+
+@jobs.background_job
+def submit_to_moss(moss_id=None, file_regex="*", assignment_id=None, language=None):
+    logger = jobs.get_job_logger()
+    logger.info('Starting MOSS Export...')
+
+    assign = Assignment.query.filter_by(id=assignment_id).one_or_none()
+    if not assign:
+        logger.info("Could not find assignment")
+        return
+
+    subms = assign.course_submissions(include_empty=False)
+    subm_keys = set(s['backup']['id'] for s in subms)
+
+    seen = set()
+    for subm in subms:
+        if subm['backup']['id'] in seen:
+            continue
+        else:
+            seen.add(subm['backup']['id'])
+
+        if subm['group']:
+            logger.info("{} -> {}".format(encode_id(subm['backup']['id']),
+                                          subm['group']['group_member_emails']))
+        else:
+            logger.info("{} -> {}".format(encode_id(subm['backup']['id']),
+                                          subm['user']['email']))
+
+    backup_query = (Backup.query.options(db.joinedload('messages'))
+                          .filter(Backup.id.in_(subm_keys))
+                          .order_by(Backup.created.desc())
+                          .all())
+
+    logger.info("Retreived {} final submissions".format(len(subm_keys)))
+
+    # TODO: Customize the location of the tmp writing (especially useful during dev)
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        # Copy in the moss script
+        with open('server/jobs/moss-submission.pl', 'r') as f:
+            moss_script = f.read()
+            moss_script = moss_script.replace('YOUR_USER_ID_HERE', str(moss_id))
```

Comment:
Nit: you don't need the `.pl` file open for the next three lines, so you can dedent them